src/Sylius/Bundle/AdminBundle/Resources/config/routing/ajax/product_variant.yml

```
@@ -37,6 +37,7 @@ sylius_admin_ajax_all_product_variants_by_phrase:
                 arguments:
                     phrase: $phrase
                     locale: expr:service('sylius.context.locale').getLocaleCode()
+                    limit: 5
```

Why 5? we didn't decide how big this limit should be but in other places we have a limit of 25. The shop can offer many variants so IMO 5 is a bit too small value :)

---

It's indeed small, 5 often won't even cover the variants of a single product.
Besides that, maybe it's better define a parameter for it, so it's easier to customize.

---

I wanted to make it dynamic but for now only 1 variable is supported in this configuration with js included. IIRC you can make it work by manually passing parameter to URL but doing so programatically is a task beyond me.
I'll make it 25 then.

---

I meant to use a parameter for the default parameter, like this `limit: '%sylius_admin.some_parameter%'`.
But what you suggest is even better and it's actually not that complicated to do:
Try this:
```suggestion
                    limit: $limit
        limit: 25
```
And to test it's overridable:
```twig
{{ path('sylius_admin_ajax_all_product_variants_by_codes', {'limit': 10}) }}
```

---

Moreover, I made this function's `public function findByPhrase(string $phrase, string $locale, ?int $limit): array` `$limit` parameter untyped when I created pagination because whatever comes to these parameters are strings :/
Now if I make `limit: $limit` then passing it to URL will cause a type error.

---

No worries, there's a feature for that in Sylius: `limit: '!!int $limit'`

---

It's not a documented feature though... 🤷‍♂️ 
/cc @loic425 @Roshyo maybe you would be interested in this one.

---

So no dynamic parameter after all? 😞

---

No because it would be wiser to keep it hardcoded as everywhere so far and if it comes the time we could make all limits accessible in twig. Dynamic limit seems more likely to be used in pagination but I understand the need of having just limit dynamic. Since my pagination went to trash I decided that dynamic limit would be hardcoded because it would be "used only in one place, so it seems to do more harm than good".

---

@rafalswierczek our mission is making Sylius as easy to customize as possible (especially if it needs only a few lines of code 👯‍♀️  , so I suggest bringing up this topic to our daily and make a small additional discussion on audio :)

---

It will be dynamic